content/en/blog/trust-chain/index.md

```diff
@@ -0,0 +1,362 @@
+> [WebAuthn Resource List](https://github.com/herrjemand/awesome-webauthn#faq)
+
+We started our tests with the new agent API by using implementing our *FIDO2
+server* and by using only browsers at the beginning. When results, especially
+the performance and simplicity, were so good, we decided to go further.
+
+The following architecture-drawing present the final deployment diagram of the
+overall system. The FIDO2 components that we implemented by ourselves are marked
+in red. The basic idea was to have a system-level SSO where we implemented
+authorization with JWT and authentication with FIDO2 regardless of which type of
+the entity needs to be authenticated: individuals, organizations, legal
+entities, or system components. For us, it implicated that we needed FIDO2 for
+service agents, which meant that a *headless* FIDO2 Authenticator was required.
+
+{{< imgproc cover-deployment.png Resize "1591x" >}}
+<em>All Key Components of The System Architecture</em>
+{{< /imgproc >}}
+
+Architectural requirements for the solution were quite complex because we wanted
+to have security covered, not to compromise performance, and still support
+polyglot development.
+
+## Polyglot Authenticator Interface
+
+FIDO2/WebAuthn specification gives a well over a description of how main
+components work. Here we focus on the two most important ones. The first is the
+authenticator registration flow which is presented picture below.
+
+![WebAuthn Registration](https://www.w3.org/TR/webauthn/images/webauthn-registration-flow-01.svg)
+<p align = "center"> FIDO2 Authenticator Registration - www.w3.org </p>
+
+To summarise, the above flow registers a new instance of an authenticator. Then
+it verifies that the same authenticator is bound to the account. That is done
+using a unique public/private key pair where the private key is in the
+authenticator. Note that the authenticator doesn't map a particular user to an
+account. That is done thru the other process flow and by the [relying
+party](https://www.w3.org/TR/webauthn/#webauthn-relying-party).
+
+The flow below shows how a registered authenticator is used to authenticate the
+account holder.
+
+![WebAuthn Authentication](https://www.w3.org/TR/webauthn/images/webauthn-authentication-flow-01.svg)
+<p align = "center"> FIDO2 Authentication - www.w3.org </p>
+
+The Command pattern was the perfect solution for the first implementation
+because it supported all of our use cases, but same time was simplest. Most
+straightforward to integration was naturally with a programming language it was
+implemented which was Go.
+
+The second thing was to figure out how we would like to implement interprocess
+communication. For that, the command pattern is suited very well. Fill the
+command with all the needed data and give one of the operations we were
+supporting: `register` and `login` from the FIDO2 standard. The process
+communication is handled just as the process starts by reading the command from
```

Comment:
For Node.js-wrapper, we give the params as CLI arguments, not as JSON 🤔

Reply:
as talking about during the meeting, because the pipe/filter is a working method, let's leave it so.